aquacomputer_d5next.c

```diff
@@ -1658,24 +1666,14 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 					return -EINVAL;
 				}
 
-				ctrl_mode += channel;
+				if (val > 0)
```

Comment:
The switch above will return -EINVAL already, this check seems unneeded.

Reply:
This check prevents adding the channel to `ctrl_mode` for the `AQUAERO_CTRL_PRESET_ID_AUTO`.
I'll revise this to make it better to understand.